public/components/trace_analytics/components/common/plots/service_map.tsx

```diff
@@ -404,14 +366,16 @@ export function ServiceMap({
     const max = Math.max(...values);
     const calculatedTicks = calculateTicks(min, max);
     setTicks(calculatedTicks);
+    // Adjust graph rendering logic to ensure related services are visible
+    const showRelatedServices = focusedService ? true : filterByCurrService;
     setItems(
       getServiceMapGraph(
         serviceMap,
         idSelected,
         calculatedTicks,
-        currService,
+        focusedService ?? currService,
         serviceMap[currService!]?.relatedServices,
-        filterByCurrService
+        showRelatedServices
       )
     );
   }, [serviceMap, idSelected]);
```

Comment:
do we need `focusedService` and `filterByCurrService` in dependency array?

Reply:
I don't believe so

https://github.com/user-attachments/assets/d97625b4-69e0-4390-a828-a088c6d682da

Reply:
We should include every dependency in the dependency array even if it seems to pointlessly re-render as to avoid 'stale' states that might lead to bugs that would be difficult to fix.
React useEffect doc: https://react.dev/reference/react/useEffect